src/ballet/block/test_microblock.c

```diff
@@ -0,0 +1,148 @@
+  /* Ensure footprint is multiple of align */
+
+  for( ulong i=0UL; i<64UL; i++ ) {
+    FD_TEST( ( fd_microblock_footprint( i ) % alignof(fd_microblock_t) )==0UL );
+  }
+
+  /* Test overflowing txn_max_cnt */
+
+  FD_TEST( fd_microblock_footprint( 5167155202719762UL )==0xfffffffffffffbc0UL );
+  FD_TEST( fd_microblock_footprint( 5167155202719763UL )==0UL  ); /* invalid */
+  FD_TEST( fd_microblock_new      ( (void *)64UL, 5167155202719763UL )==NULL ); /* fake shmem, invalid footprint */
+
+  /* Test failure cases for fd_microblock_new */
+
+  FD_TEST( fd_microblock_new( NULL,        16UL )==NULL ); /* null shmem */
+  FD_TEST( fd_microblock_new( (void *)1UL, 16UL )==NULL ); /* misaligned shmem */
+
+  /* Test failure cases for fd_microblock_join */
+
+  FD_TEST( fd_microblock_join( NULL        )==NULL ); /* null shblock */
+  FD_TEST( fd_microblock_join( (void *)1UL )==NULL ); /* misaligned shblock */
+
+  /* Test microblock creation */
+
+  ulong txn_max_cnt = 16UL;
+  ulong footprint = fd_microblock_footprint( txn_max_cnt );
+  FD_TEST( footprint );
+
+  void * shmem = fd_alloca( FD_MICROBLOCK_ALIGN, footprint );
+  FD_TEST( shmem );
+
+  void * shblock = fd_microblock_new( shmem, txn_max_cnt );
+
+  fd_microblock_t * block = fd_microblock_join( shblock );
+  FD_TEST( block );
+
+  /* Test bad magic value */
+
+  block->magic++;
+  FD_TEST( fd_microblock_join( shblock )==NULL );
+  block->magic--;
+
+  /* Test microblock destruction */
+
+  FD_TEST( fd_microblock_leave( NULL  )==NULL    ); /* null block */
+  FD_TEST( fd_microblock_leave( block )==shblock ); /* ok */
+
+  FD_TEST( fd_microblock_delete( NULL        )==NULL ); /* null shblock */
+  FD_TEST( fd_microblock_delete( (void *)1UL )==NULL ); /* misaligned shblock */
+
+  /* Test bad magic value.
+     Note that at this point our `block` pointer is dangling. */
+  block->magic++;
+  FD_TEST( fd_microblock_delete( shblock )==NULL );
+  block->magic--;
+
+  FD_TEST( fd_microblock_delete( shblock )==shmem );
+}
+
+/* A serialized batch of entries.
+   Sourced from the genesis of a `solana-test-validator`. */
+FD_IMPORT_BINARY( localnet_batch_0, "src/ballet/shred/fixtures/localnet-slot0-batch0.bin" );
+
+/* Target buffer for storing an `fd_microblock_t`.
+   In production, this would be a workspace. */
+static uchar __attribute__((aligned(FD_MICROBLOCK_ALIGN))) microblock_buf[ 0x40000 ];
+
+struct fd_microblock_test_vec {
+  uchar mixin[ FD_SHA256_HASH_SZ ];
+  fd_microblock_hdr_t hdr;
+};
+typedef struct fd_microblock_test_vec fd_microblock_test_vec_t;
+
+void
+test_parse_localnet_batch_0( void ) {
+  FD_TEST( localnet_batch_0_sz==3080UL );
+
+  /* Peek the number of entries, which is the first ulong. */
+  ulong microblock_cnt = *(ulong *)localnet_batch_0;
+  FD_TEST( microblock_cnt==64UL );
+
+  /* Move past the first ulong to the entries. */
+  void * batch_buf = (void *)((uchar *)localnet_batch_0    + 8UL);
+  ulong  batch_bufsz =                 localnet_batch_0_sz - 8UL ;
```

Comment:
I was surprised to see you manually skipping some bytes until I realized the fixture is actually a batch containing many microblocks not a single microblock. It might be worth throwing in a sentence along those lines.